### ISSUE TITLE
Compatibility with pallet-contracts?

### ISSUE THREAD

**Issue by joshuacheong** [AcalaNetwork/bodhi.js#669]
- Would bodhi.js be compatible with [pallet-contracts](https://github.com/paritytech/substrate/tree/master/frame/contracts)?
- If not, what would it take to help WASM smart contracts (even if they are compile using Solang from Solidity) benefit from Ethereum tooling (Hardhat/Truffle) and wallets (like Metamask)?

**Comment by xlc** [comment]
It is not compatible. We will need to do some research to identify the feasibility but it is definitely not outside the realm of the possible.

**Comment by joshuacheong** [comment]
So I am trying to understand more about Acala, so EVM+ is not using frontier and it is fully custom?

**Comment by xlc** [comment]
Yes we are not using frontier. The [evm engine](https://github.com/rust-blockchain/evm/) is the same but otherwise it is fully custom with many new features. You can read more here: https://evmdocs.acala.network/general/about-acala-evm+
